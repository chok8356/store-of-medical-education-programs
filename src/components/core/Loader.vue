<template>
  <div class="loader">
    <b-spinner
      label="Loading..."
      variant="primary" />
  </div>
</template>

<script lang="ts">
import { BSpinner } from 'bootstrap-vue'
import { Vue, Component } from 'vue-property-decorator'

@Component({
  name: 'Loader',
  components: {
    BSpinner
  }
})
export default class Loader extends Vue {
}
</script>

<style lang="scss">
.loader {
  align-items: center;
  background-color: rgba(#fff, 0.75);
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  opacity: 1;
  position: absolute;
  top: 0;
  transition: opacity 0.1s ease-in;
  width: 100%;
  z-index: 1;
}

.loader-parent {
  &--relative {
    position: relative;
  }
}
</style>
