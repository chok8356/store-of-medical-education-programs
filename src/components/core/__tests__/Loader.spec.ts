import { shallowMount } from '@vue/test-utils'
import { BSpinner } from 'bootstrap-vue'
import Loader from '../Loader.vue'

describe('Loader', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Loader)
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('render correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it('spinner props', () => {
    const spinnerComponent = wrapper.findComponent(BSpinner)
    expect(spinnerComponent.attributes('label')).toBe('Loading...')
    expect(spinnerComponent.attributes('variant')).toBe('primary')
  })
})
