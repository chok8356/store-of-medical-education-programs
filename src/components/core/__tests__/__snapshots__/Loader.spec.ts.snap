// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader render correctly 1`] = `
<div
  class="loader"
>
  <b-spinner-stub
    label="Loading..."
    role="status"
    tag="span"
    type="border"
    variant="primary"
  />
</div>
`;
